src/main/java/org/folio/search/service/ResourceService.java

```diff
@@ -129,20 +129,26 @@ public FolioIndexOperationResponse indexConsortiumInstances(List<ConsortiumInsta
       return getSuccessIndexOperationResponse();
     }
 
-    var instanceIdsByTenant = consortiumInstances.stream()
-      .collect(groupingBy(ConsortiumInstanceEvent::getTenant,
-        Collectors.mapping(ConsortiumInstanceEvent::getInstanceId, toSet())));
+    var validConsortiumInstances = consortiumInstances.stream()
+      .filter(event -> consortiumTenantService.getCentralTenant(event.getTenant()).isPresent())
+      .distinct()
+      .toList();
 
-    Map<String, List<SearchDocumentBody>> allResourceEvents = new HashMap<>(consortiumInstances.size());
-    for (var tenantInstanceIds : instanceIdsByTenant.entrySet()) {
-      tenantScopedExecutionService.executeTenantScoped(tenantInstanceIds.getKey(), () -> {
-        var resourceEvents = consortiumInstanceService.fetchInstances(tenantInstanceIds.getValue());
-        allResourceEvents.putAll(multiTenantSearchDocumentConverter.convert(resourceEvents));
-        return null;
-      });
+    if (log.isDebugEnabled()) {
+      var invalidInstances = ListUtils.subtract(consortiumInstances, validConsortiumInstances);
+      log.debug("Skip indexing consortium instances [{}]", invalidInstances);
     }
 
-    return indexSearchDocuments(allResourceEvents);
+    var centralTenant = consortiumTenantService.getCentralTenant(validConsortiumInstances.get(0).getInstanceId())
```

Comment:
Why is instanceId passed here?
Are you sure /user-tenants will return central tenant in such case?